### DISCUSSION TITLE
Feedback needed on new UI - Bulk Editing on Prims

### DISCUSSION THREAD

**Discussion by womanyoyoyo-adsk** [Autodesk/maya-usd#3155] [Show and tell]
Hello,

We have new design that we'd like to gather feedback about. This design allows users to run USD actions on prims in bulk. Please watch the attached video to see it.

https://github.com/Autodesk/maya-usd/assets/49446314/1a48eaaa-33dd-4c8a-9815-77017b6a9882

We would love to hear your feedback on this! Some questions:

1. What USD actions do you need to run on multiple prims?
2. Do you think the [In-View Editor](https://help.autodesk.com/view/MAYAUL/2024/ENU/?guid=GUID-C9C88AA3-CE61-4389-9A19-B6456427EBA2) is the correct piece of UI for switching variants quickly?
3. Any other thoughts? 

Thank you,
Natalia 

PS. You can also email me personally with feedback at natalia.minaki@autodesk.com

**Comment by swillisart** [comment]
> What USD actions do you need to run on multiple prims?

    - Variant Set picking.
    - Material Assignment.
    - For Maya data "Move to USD stage" as in addition to "Duplicate As USD Data" since duplication won't scale very well in bulk complex scenes.

> Do you think the [In-View Editor](https://help.autodesk.com/view/MAYAUL/2024/ENU/?guid=GUID-C9C88AA3-CE61-4389-9A19-B6456427EBA2) is the correct piece of UI for switching variants quickly?

I think it fits pretty well. Would be extra useful if the tool can be made to react to changes in selection even after the tool is launched and persistently open, that's a great re-use of Maya's model editing and will be familiar to everyone.

The only possible very small downside to this design (which may never actually occur) is if the author makes an asset that has an excessive number of variants. For example: a rivet asset that encodes each nut or bolt as a variation of some compound screw. In that case you would want some sort of keyword search filter and a scrollable list view.

> Any other thoughts?

We will want to make our own custom actions based on the selected prim types (meshes, cameras, curves) so be extra sure to preserve Maya's menu callback hook for first-party and third-party custom actions.

I would also like to update my "USD Selection Mode" from the context menu but that is non-essential if I can simply add it myself.

Thats all the feedback I've got,
Thanks!

**Reply by womanyoyoyo-adsk** [reply]
Thanks @swillisart for all this great feedback. We are looking at this now. For the 'USD Selection Mode', you can tear it off and have it as a floating window if you click on the dots at the top of the submenu: 

![image](https://github.com/Autodesk/maya-usd/assets/49446314/a42010ae-cdb6-4b6c-b996-ee4c7de392dd)

**Comment by sharktacos** [comment]
In addition to the suggestions of @swillisart, which I'd second, I'll add that I'd appreciate having  "mark as instanceable" included in the bulk actions as @womanyoyoyo-adsk suggested in the vid.

**Comment by womanyoyoyo-adsk** [comment]
@sharktacos Ok great to know. I think it will have to have Unmark as Instanceable too.

**Reply by sharktacos** [reply]
That makes sense. Something to consider additionally is having "make instanceable" as a settings option on the **Add USD reference...** dialog. 

<img width="923" alt="image" src="https://github.com/Autodesk/maya-usd/assets/23034637/72fbf58f-c7cc-4cfc-8bef-8ce60dfe300d">


Then a user could simply set the option as desired and have it come in that way when creating the reference. Saves a step in the workflow.
